yourgov/components/FormDefinitionList.tsx

```diff
@@ -1,24 +1,116 @@
-import { ReactNode } from 'react';
+import { Fragment, PropsWithChildren } from 'react';
+import format from 'date-fns/format';
+import parseISO from 'date-fns/parseISO';
 import {
 	SummaryList,
 	SummaryListItem,
 	SummaryListItemDescription,
 	SummaryListItemTerm,
 } from '@ag.ds-next/react/summary-list';
 
-export function FormDefinitionList({
-	items,
-}: {
-	items: { label: string; value: ReactNode }[];
-}) {
+export function FormDefinitionList(props: PropsWithChildren<{}>) {
+	return <SummaryList>{props.children}</SummaryList>;
+}
+
+type FormDefinitionListItemProps = {
+	label: string;
+	value: string | undefined;
+};
+
+export function FormDefinitionListItem({
+	label,
+	value,
+}: FormDefinitionListItemProps) {
+	return (
+		<SummaryListItem>
+			<SummaryListItemTerm>{label}</SummaryListItemTerm>
+			<SummaryListItemDescription>{value}</SummaryListItemDescription>
+		</SummaryListItem>
+	);
+}
+
+type FormDefinitionListItemAddressProps = {
+	label: string;
+	address: string | undefined;
+	suburb: string | undefined;
+	state: string | undefined;
+	postcode: string | undefined;
+};
+
+export function FormDefinitionListItemAddress({
+	label,
+	address,
+	suburb,
+	state,
+	postcode,
+}: FormDefinitionListItemAddressProps) {
+	return (
+		<SummaryListItem>
+			<SummaryListItemTerm>{label}</SummaryListItemTerm>
+			<SummaryListItemDescription>
+				{!address || !suburb || !state || !postcode
+					? null
+					: `${address}, ${suburb} ${state} ${postcode}`}
+			</SummaryListItemDescription>
+		</SummaryListItem>
+	);
+}
+
+type FormDefinitionListItemDateProps = {
+	label: string;
+	value: unknown;
+};
+
+export function FormDefinitionListItemDate({
+	label,
+	value,
+}: FormDefinitionListItemDateProps) {
+	return (
+		<SummaryListItem>
+			<SummaryListItemTerm>{label}</SummaryListItemTerm>
+			<SummaryListItemDescription>
+				{!value ? null : (
```

Comment:
I'm always intrigued why people check the negative (which is arguable more code and potentially minisculely slower, than just doing `value ? <Fragment /> : null`

Reply:
Although rare, I do think sometimes  there are some situations where it is easier to read the negative. But this is not one of those situations, so I've updated all of the ternary operators in this file to make it easier to read.